src/Microsoft.Security.Utilities.Benchmarks/SecretMaskerDetectionBenchmarks.cs

```diff
@@ -5,42 +5,85 @@
 
 namespace Microsoft.Security.Utilities.Benchmarks
 {
-    public abstract class SecretMaskerDetectionBenchmarks
+    public enum RegexEngine
     {
-        // The size of randomized data to add as a prefix
-        // for every secret. This is intended to make positive
-        // hit less concentrated in the profiling.
-        private const int SecretPrefixSize = 100 * 1024;
+        DotNet,
+        RE2,
+    }
+
+    public enum Case
+    {
+        // This is close to the worst case scenario for large input. Every
+        // base64 alphabet char is equally likely to appear in the input, making
+        // signature sniffing harder.
+        HardPrefix,
+
+        // This is close to the best case scenario. A char that does not appear
+        // in any identifiable key signature is repeated in the prefix, making
+        // signature sniffing easier.
+        EasyPrefix,
+
+        // This will scan secrets without prefixing them. Testing the performance against
+        // small inputs.
+        NoPrefix,
+    }
 
-        protected List<string> RandomExamples { get; }
-        protected List<string> NonRandomExamples { get; }
+    public abstract class SecretMaskerDetectionBenchmarks
+    {
+        // The size of randomized data to add as a prefix for every secret for
+        // the EasyPrefix and HardPrefix cases. This is intended to make
+        // positive hits less concentrated in the profiling.
+        public const int SecretPrefixSize = 10 * 1024;
```

Comment:
>1024 [](http://example.com/codeflow?start=49&length=4)

ok, so default filler material is 10K? that still seems awfully concentrated compared to a real world scenario. Also this assumes this number is `InBytes` (you might consider adding an explanatory size suffix to the name, I always find this useful).

Reply:
It's InChars, technically. 

It was 100KB, but it's painful to iterate on the benchmark when testing that much.